internal/proposals/improvement-of-log-messages-usability.md

```diff
@@ -0,0 +1,75 @@
+- context - structure of a text contextual information such as operation (for example: "starting workers"), handler/ resolver (for example: "ProvisionRuntime"), controller, resource namespaced name (for example: "production/application1", operation ID, instance ID, operation stage and so on. User must be able to filter the logs that are needed so all the info provided here must be useful and be an unique minimal set for every operation. User must be able to find the needed resource in some store so provide here a name instead of an ID if it's easier to use later
+- trace_id - 16-byte numeric value as Base16-encoded string. It'll be passed through a header, so the user can filter all the logs regarding the whole business operation in the whole system
+- span_id - 16-byte numeric value as Base16-encoded string. It'll be randomly generated for each request handling so the user can filter the component logs for a specific operation handling
+> trace_id and span_id are required in the log to be compliant with the [OpenTelemetry standards](https://github.com/open-telemetry/oteps/pull/114/files)
+
+### Log format examples
+
+#### Key-Value Pairs
+```text
+2020-12-15T07:26:45+00:00 WARNING Tiller configuration not found in the release artifacts. Proceeding to the Helm 3 installation... context.resolver=ProvisionRuntime context.operationID=92d5d8fd-cbdc-4b7a-9bc3-2b2eccfcb109 context.stage=InstallReleaseArtifacts context.shootName=c-3a38b3a context.runtimeId=19eb9335-6c13-4d40-8504-3cd07b18c12f trace_id=0354af75138b12921 span_id=14c902d73a
+```
+
+#### JSON
+```json
+{
+  "timestamp": "2020-12-15T07:26:45+00:00",
+  "level": "WARNING",
+  "message": "Tiller configuration not found in the release artifacts. Proceeding to the Helm 3 installation...",
+  "context": {
+    "resolver": "ProvisionRuntime",
+    "operationID": "92d5d8fd-cbdc-4b7a-9bc3-2b2eccfcb109",
+    "stage": "InstallReleaseArtifacts",
+    "shootName": "c-3a38b3a",
+    "runtimeID": "19eb9335-6c13-4d40-8504-3cd07b18c12f"
+  },
+  "trace_id": "0354af75138b12921",
+  "span_id": "14c902d73a"
+}
+```
+
+## What should **not** be logged
+
+We should be careful about the information we put in logs. Every log should be easily connectable with the issue or the error but there should be no internal or confidential data provided in any log entry. Here is what should not be logged:
+
+- authorization data such as passwords, usernames, certificates and tokens. Note that sometimes basic auth could be used in the url such as `http://username:password@example.com/` and other tokens like `http://www.example.com/api/v1/users/1?access_token=123` so you need to be extra careful in these cases
+- personal identifiable information such as tenant names, sub account IDs, global account IDs and so on. If the personal identifiable information is required in the audit log it should be encrypted
```

Comment:
Did you confirm that subaccount and globalaccount IDs are going into this category? There were no objections to logging those in prior approaches. I would like to have a clear statement on this as it will affect most of our production components (KEB, Provisioner, whole Compass, etc.)

Reply:
I'll ask security experts specifically about those two

Reply:
Asked and it should be fine